src/main/java/com/internship/backend/service/TennisCourtService.java

```diff
@@ -26,12 +25,12 @@ public List<TennisCourt> getAllTennisCourts(){
     }
 
     public TennisCourt addTennisCourt(TennisCourt tennisCourt) throws TennisCourtAlreadyExistsException {
-        for (Location elem : locationRepository.findAll()){
-            if (elem.getId() == tennisCourt.getLocation().getId()){
-                if (!elem.getTennisCourt().contains(tennisCourt)) {
-                    List<TennisCourt> aux = elem.getTennisCourt();
-                    aux.add(tennisCourt);
-                    elem.setTennisCourt(aux);
+        for (Location location : locationRepository.findAll()){
+            if (location.getId() == tennisCourt.getLocation().getId()){
```

Comment:
Could we define a private method responsible for finding a location that has the tennisCourt location id? 
Something like: 
private Optional<Location> findLocation(int locationId) {
locationRepository.findAll()
                .stream()
                .filter(l -> l.getId() == locationId)
                .findFirst();
}